File: Go/countNegatives.go
package main

// O(N+M)
// O(1)

// If your matrix does not have a specific order or you do not know its structure, use the nested loops method. N * M
// If the rows of your matrix are sorted, the method starting from the top right corner. N + M

func countNegatives(grid [][]int) int {
	count := 0
	row, col := len(grid), len(grid[0])
	i, j := 0, col-1

	for i < row && j >= 0 {
		if grid[i][j] < 0 {
			count += row - i
			j--
		} else {
			i++
		}
	}
	return count
}

File: C#/CountNegatives.cs
//O(n^2)
//O(1)
public class Solution
{
    public int CountNegatives(int[][] grid)
    {
        if (grid.Length == 0 || grid[0].Length == 0) return 0;

        int count = 0;

        for (int i = 0; i < grid.Length; i++)
        {
            for (int j = 0; j < grid[i].Length; j++)
            {
                if (grid[i][j] < 0)
                {
                    count++;
                }
            }
        }

        return count;
    }
}